src/services/outliningElementsCollector.ts

```diff
@@ -35,6 +44,19 @@ namespace ts.OutliningElementsCollector {
             }
         }
 
+        function addOutliningSpanRegions(regionSpan: RegionRange) {
+            if (regionSpan) {

```

Comment:
why not just create OutliningSpans for every region when we are building them instead of creating the extra object. less garbage to collect later.

Reply:
Fixed.